lib/topological_inventory.rb

```diff
@@ -16,6 +20,7 @@ def self.call
     TopologicalInventoryApiClient.configure do |config|
       config.host = ENV['TOPOLOGICAL_INVENTORY_URL'] || 'localhost'
       config.scheme = URI.parse(ENV['TOPOLOGICAL_INVENTORY_URL']).try(:scheme) || 'http'
+      config.base_path = File.join("/", ENV['PATH_PREFIX'].presence || DEFAULT_PATH_PREFIX, SERVICE_NAME, VERSION)
```

Comment:
:heart:

Reply:
@carbonin 
The gem is directly built off the openapi.json so it has the base path built into it.
Say we ship with Version 1 of the Gem we would set the base_path here to /api/topo/v1 and that manages the default value in the Gem.

Now we add new features and change these environment variables to be v2 so the base_path is now at runtime switches to /api/topo/v2

Still using the Gem that was bundled in the Gemfile for catalog v1, that gem doesn't know how to handle v2 requests and it would fail.

If you rebuild the Gem the base path would be correct, we can't expect v2 features to be available in the v1 gem, the v2 objects could be completely different.

Reply:
Base path and versioning are different concerns in my mind. If you want to parse out the version from the default path and use that unconditionally, that's fine with me. But I think that we should not be using the base path as defined in the gem as that should be defined entirely at the platform/configuration level. The client gem should never make assumptions about how the server is configured.